### PR TITLE
[vSphere][network] Create network datastream

### PR DESCRIPTION
**Pull request by lucian-ioan**
## Proposed commit message

Following beats code changes in https://github.com/elastic/beats/pull/40559, the following metrics are added to the network datastream:

Metrics Type | Metrics | API | Field Mappings
-- | -- | -- | --
Network/DistributedVirtualPort | Network.OverallStatus | Summary | status
  | Network.Summary.Accessible | Summary | accessible
  | Network.ConfigStatus | Summary | config.status
  | Network.type | Summary | type
  | Network.Vm | ExtraApi | vm.names
  |   | ExtraApi | vm.count
  | Network.Host | ExtraApi | host.names
  |   | ExtraApi | host.count

Alerts |  
-- | -- 
triggerd_alarms.name | Name of the alarm.
triggerd_alarms.id | Unique key that identifies the alarm.
triggerd_alarms.status | Overall status of the alarm object.
triggerd_alarms.triggered_time | Time the alarm triggered.
triggerd_alarms.description | Description of the alarm.
triggerd_alarms.entity_name | alarm happened in respective entity

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Author's Checklist

- [x] Create Snapshot of Metricbeat changes.
- [x] Use Snapshot to make the changes in the vSphere Integration Host datastream.
- [x] Update the system and pipeline changes.
- [x] Add versioning to changelog.
- [x] Test the Integration for metric collection.
- [x] Test Integration from TSDB Perspective.
- [x] Code review.

## Related issues

-  Closes https://github.com/elastic/obs-integration-team/issues/40

## TSDB testing
```
Values being used:
        elasticsearch_host = https://localhost:9200
        elasticsearch_ca_path = /home/lucian/.elastic-package/profiles/default/certs/elasticsearch/ca-cert.pem
        elasticsearch_user = elastic
        elasticsearch_pwd = changeme
        data_stream = metrics-vsphere.network-default
        docs_index = First index of the data stream
        settings_mappings_index = Last index of the data stream
        max_docs = All documents
        get_overlapping_files = True
        directory_overlapping_files = overwritten-docs-metrics-vsphere.network-default
        display_docs = 10
        copy_docs_per_dimension = 2

You're testing with version 8.15.2-SNAPSHOT.

Testing data stream metrics-vsphere.network-default.
Index being used for the documents is .ds-metrics-vsphere.network-default-2024.09.23-000001.
Index being used for the settings and mappings is .ds-metrics-vsphere.network-default-2024.09.23-000001.

The time series fields for the TSDB index are: 
        - dimension (9 fields):
                - agent.id
                - cloud.account.id
                - cloud.availability_zone
                - cloud.instance.id
                - cloud.provider
                - cloud.region
                - container.id
                - host.name
                - service.address
        - gauge (2 fields):
                - vsphere.network.host.count
                - vsphere.network.vm.count
        - routing_path (9 fields):
                - agent.id
                - cloud.account.id
                - cloud.availability_zone
                - cloud.instance.id
                - cloud.provider
                - cloud.region
                - container.id
                - host.name
                - service.address

Index tsdb-index-enabled successfully created.

Copying documents from .ds-metrics-vsphere.network-default-2024.09.23-000001 to tsdb-index-enabled...
All 12 documents taken from index .ds-metrics-vsphere.network-default-2024.09.23-000001 were successfully placed to index tsdb-index-enabled.
```

### CODE REVIEW

File: packages/vsphere/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "1.14.0-next"
```
Comment (ishleenk17) [inline packages/vsphere/changelog.yml:2]: change version to 1.15.0-next

File: packages/vsphere/data_stream/network/fields/fields.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      fields:
+        - name: names
+          type: keyword
+          description: >
+            Names of the virtual machines connected to this network.
+        - name: count
+          type: long
+          metric_type: gauge
+          description: >
+            Number of virtual machines connected to this network.
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/network/fields/fields.yml:49]: Missing the latest fields

**Comment by ishleenk17** [comment]
1. README is missing. 
2. update kibana version
3. Perios has not been taken care in this

File: packages/vsphere/data_stream/network/manifest.yml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+title: "vSphere network metrics"
+type: metrics
+streams:
+  - input: vsphere/metrics
+    title: vSphere network metrics
+    description: vSphere network metrics
```
Comment (niraj-elastic) [inline packages/vsphere/data_stream/network/manifest.yml:6]: please add `period` here, along with other configuration parameters.

File: packages/vsphere/data_stream/network/agent/stream/stream.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,10 @@
+metricsets: ["network"]
+hosts:
+{{#each hosts}}
+  - {{this}}
+{{/each}}
+period: {{period}}
+username: {{username}}
+password: {{password}}
+# If insecure is true, don't verify the server's certificate chain
+insecure: {{insecure}}
```
Comment (niraj-elastic) [inline packages/vsphere/data_stream/network/agent/stream/stream.yml.hbs:10]: please add `tags` and `processors`.

**Review by niraj-elastic — CHANGES_REQUESTED** [review]

File: packages/vsphere/changelog.yml [hunk 5]
```diff
@@ -1,10 +1,13 @@
 # newer versions go on top
 - version: "1.15.0-next"
   changes:
+    - description: Add new network datastream.
+      type: enhancement
+      link: https://github.com/elastic/integrations/pull/11205
```
Comment (ishleenk17) [inline packages/vsphere/changelog.yml:6]: Incorrect PR number

File: packages/vsphere/data_stream/network/fields/agent.yml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+      type: keyword
+      dimension: true
+      ignore_above: 1024
+      description: 'Name of the host.  It can contain what `hostname` returns on Unix systems, the fully qualified domain name, or a name specified by the user. The sender decides which value to use.'
+    - name: containerized
+      type: boolean
+      description: >
+        If the host is a container.
+
+    - name: os.build
```
Comment (ishleenk17) [inline packages/vsphere/data_stream/network/fields/agent.yml:18]: These are ecs fields w/o dimensions. Not required here.

**Comment by ishleenk17** [comment]
@lucian-ioan : Pls share details of TSDB testing

File: packages/vsphere/changelog.yml [hunk 7]
```diff
@@ -1,10 +1,13 @@
 # newer versions go on top
 - version: "1.15.0-next"
```
Comment (kush-elastic) [inline packages/vsphere/changelog.yml:2]:
```suggestion
- version: "1.15.0"
```
As this PR is the last one we need to remove `next` tag and update version in manifest to `1.15.0`.

File: packages/vsphere/data_stream/network/_dev/test/pipeline/test-network.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+                  "accessible": true,
+                  "config": {
+                    "status": "green"
+                  },
+                  "name": "VM Network",
+                  "status": "green",
+                  "type": "Network"
+                }
+              }
+
```
Comment (kush-elastic) [inline packages/vsphere/data_stream/network/_dev/test/pipeline/test-network.json:15]:
```suggestion
```

Empty Line

File: packages/vsphere/changelog.yml (hunk 7 above)
Comment (lucian-ioan) [reply packages/vsphere/changelog.yml:2]: I guess so, should we do it with this PR @ishleenk17?

File: packages/vsphere/data_stream/network/sample_event.json [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+            "kernel": "3.10.0-1160.118.1.el7.x86_64",
+            "name": "Ubuntu",
+            "platform": "ubuntu",
+            "type": "linux",
+            "version": "20.04.6 LTS (Focal Fossa)"
+        }
+    },
+      "metricset": {
+        "name": "network",
+        "period": 20000
```
Comment (kush-elastic) [inline packages/vsphere/data_stream/network/sample_event.json:56]: This formatting seems wrong, can you check once?
```suggestion
    },
    "metricset": {
      "name": "network",
      "period": 20000
```

**Review by kush-elastic — CHANGES_REQUESTED** [review]
Added few comments

File: packages/vsphere/data_stream/network/fields/fields.yml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+      type: keyword
+      description: >
+        Type of the network (e.g., Network(Standard), DistributedVirtualport).
+    - name: vm
+      type: group
+      fields:
+        - name: names
+          type: keyword
+          description: >
+            Names of the virtual machines connected to this network.
```
Comment (niraj-elastic) [inline packages/vsphere/data_stream/network/fields/fields.yml:44]:
```suggestion
            List of all the virtual machines connected to this network.
```

minor nit. Please do it for other similar fields as well.

**Review by niraj-elastic — CHANGES_REQUESTED** [review]

File: packages/vsphere/data_stream/network/sample_event.json (hunk 9 above)
Comment (lucian-ioan) [reply packages/vsphere/data_stream/network/sample_event.json]: Should work fine but re-formatted it just in case.

**Review by kush-elastic — APPROVED** [review]
LGTM!

File: packages/vsphere/docs/README.md [hunk 11]
```diff
@@ -766,6 +767,132 @@ Please refer to the following [document](https://www.elastic.co/guide/en/ecs/cur
...
+| vsphere.network.triggered_alarms.entity_name | Name of the entity associated with the alarm. | keyword |  |
+| vsphere.network.triggered_alarms.id | Unique identifier for the alarm. | keyword |  |
+| vsphere.network.triggered_alarms.name | Name of the alarm. | keyword |  |
+| vsphere.network.triggered_alarms.status | Status of the alarm. | keyword |  |
+| vsphere.network.triggered_alarms.triggered_time | Time when the alarm was triggered. | date |  |
+| vsphere.network.type | Type of the network (e.g., Network(Standard), DistributedVirtualport). | keyword |  |
+| vsphere.network.vm.count | Number of virtual machines connected to this network. | long | gauge |
+| vsphere.network.vm.names | List of all the virtual machines connected to this network. | keyword |  |
+
+
```
Comment (niraj-elastic) [inline packages/vsphere/docs/README.md:895]:
```suggestion
```

**Review by niraj-elastic — CHANGES_REQUESTED** [review]

File: packages/vsphere/docs/README.md (hunk 11 above)
Comment (lucian-ioan) [reply packages/vsphere/docs/README.md:895]: I don't think this is necessary, it's the default format for all datastreams in docs.

**Review by niraj-elastic — APPROVED** [review]

**Comment by niraj-elastic** [comment]
LGTM

File: packages/vsphere/changelog.yml (hunk 7 above)
Comment (ishleenk17) [reply packages/vsphere/changelog.yml:2]: Lets keep the next label as of now.
We will raise a PR when we are ready to release it.

**Review by ishleenk17 — APPROVED** [review]

**Pull request merged**